### ISSUE TITLE
Add controlled timeout to article data fetch

### ISSUE THREAD

**Issue by dr3** [bbc/simorgh#1304]
**Is your feature request related to a problem? Please describe.**
When fetching a story, we should add a timeout to the data fetch to ensure that users are not stranded in a loading state.

**Describe the solution you'd like**
Blocked on UX - how do we want to handle the below (message, error page, etc):

> When the fetch takes too long, throw a 408 Request Timeout error and surface an error message. The timeout value for client and server can be different to reduce the chance of network flakiness causing articles not to render on the server, while also not causing server hanging. 

**Testing notes**
[Tester to complete]

Dev insight: This can be testing by setting VERY strong network delay in chrome dev tools. The timeout value should be in the range 10+ seconds to avoid this happening simply when you have crap internet

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

**Comment by ChrisBAshton** [comment]
What happens when we get the 408? Redirect the user to the article with `window.location`? Display an error message saying the service is having issues?

**Comment by dr3** [comment]
Im not sure, it feels like we need some UX for a client time out? What does @jimjohnsonrollings think? Its almost like we need a "Unable to connect! Maybe youre offline? Please check your internet and retry"

**Comment by jimjohnsonrollings** [comment]
I think that if the client request times out (maybe after multiple attempts?) we should attempt to load the whole thing again from the server, as Chris suggests.

**Comment by ChrisBAshton** [comment]
Hmm taking a step back and thinking about _when_ we expect this scenario to happen - I'd originally read it as "there's a problem with the BBC service", rather than "there's a problem with the reader's network connection". The latter is far more likely.

So I'd suggest some sort of offline message like Drew suggested.

**Comment by jimjohnsonrollings** [comment]
How would we be able to get any sort of data on which scenario was happening?

**Comment by ChrisBAshton** [comment]
Had a chat with Benjamin. I think we're agreed that

A 500 error should fall back to a server-side render. A 200 should client-side render. A non-200 or non-500 error should fall back to a "Please try again later" kind of message, inline.

**Comment by ChrisBAshton** [comment]
~~A 5xx error should fall back to a server-side render.~~ EDIT: client-side error page.
A 200 should client-side render.
Some errors should fall back to a "Please try again later" kind of message, inline.
~~We should talk to Mozart about exactly what status codes we can expect to get from them, and make a table of expected Simorgh behaviour for each status code.~~ DONE: covered in #1314.

**Comment by ChrisBAshton** [comment]
Mozart will throw any one of the following six status codes:

In addition to 200, 404, 500:

"We do redirects too, 301 and 302, also in rare CSRF related cases it can return a 403 for the articles JSON content".

We should decide what to do in each case for Simorgh, for both server-side render and client-side render.

| Render | Mozart response | Expected behaviour |
|--------|------------------|---------------------|
| Server | 200 | Render the page |
| Client  | 200 | Render the page |
| Server  | 404 | Mozart will show their 404 page |
| Client  | 404 | We show our own 404 page |
| Server  | 500 | TBC |
| Client  | 500 | TBC |
| Server  | 301 | TBC |
| Client  | 301 | TBC |
| Server  | 302 | TBC |
| Client  | 302 | TBC |
| Server  | 403 | TBC |
| Client  | 403 | TBC |

**Comment by ChrisBAshton** [comment]
Blocked on UX design for article data fetch timeout (due to network issues on the client).

**Comment by ChrisBAshton** [comment]
Superseded by #1384 and #1383.
